### ISSUE TITLE
Conda index CLI reference page maybe should be updated to reference separate conda-index tool

### ISSUE THREAD

**Issue by PatSunter** [conda/conda-build#5235]
### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

**TL; DR:** it looks like the "conda index" docs reference page now needs to be updated to redirect users to read https://conda.github.io/conda-index/#conda-index now it's a separate project and had to be invoked differently.

**More detail:** I have been using an older conda-build version and running the `conda index` command happily until recently (3.28.2) .

However having just upgraded to conda-build package version `24.1.2` - the `conda index` command stopped being available.

Going doc-hunting on the internet the first hit I find is this project's https://docs.conda.io/projects/conda-build/en/latest/resources/commands/conda-index.html reference page - which as of today (14 Feb 2024) says to run `conda index` as I did previously.

Some extra searching and I found https://github.com/conda/conda-index?tab=readme-ov-file#conda-index, and pieced together that it's now a separate project, and my conda env update meant I had switched to using it.

As those docs currently say re how to invoke conda index when using the new package:

> python -m conda_index <path to channel directory>
> Note conda index (instead of python -m conda_index) may find legacy conda-build index.

So replacing the conda-index docs page in this repo with some redirect guidance would be nice please - especially for users like me in this `conda update conda-build` use-case.

**Comment by PatSunter** [comment]
Ah - further to this issue report, via looking at https://github.com/conda/conda-index/blob/main/CHANGELOG.md#040-2024-01-29 I see there is now supposed to be ability to run `conda index` command and invoke the new separate conda-index, per:
https://github.com/conda/conda-index/issues/81

Which back-references in this project issues/PRs #4828, #4645 - and one of those references a PR to start marking conda-index as deprecated: #4885 .

So a couple of remarks to help close this issue:

1) Re #4885 - I see it added a note into changelog:
There is a note about the deprecated behaviour showing at:
https://docs.conda.io/projects/conda-build/en/24.3.x/release-notes.html#id46

but there is no "deprecated" text showing on this page right now (feel like something warranted here): https://docs.conda.io/projects/conda-build/en/24.3.x/resources/commands/conda-build.html

2) Re why I had trouble running `conda index` after I did an upgrade:

Reading the conda-build docs again I see in this page:
https://docs.conda.io/projects/conda-build/en/latest/install-conda-build.html

... your pretty strong recommendation to install conda-build into a `base` environment.

Having just done this, I now can run `conda build` and it appears to be using the new separate conda-build package judging by the updated output I get.

So maybe not too much needs to change re these docs ?? But perhaps still some note on the conda index CLI reference page to make it clear that it's now being provided by a separate tool.
